File: remove_scrollbars.js
const fs = require('fs');
const path = require('path');

const directories = ['app', 'components'];

function processFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf8');
    let newContent = content;

    // We use a regex to find <ScrollView and <FlatList that don't already have showsVerticalScrollIndicator
    // In JS, regex lookbehind/lookahead can be tricky with variable length.
    // Let's just use string replacement on the opening tags if they don't contain it.
    
    // Replace <ScrollView ...>
    newContent = newContent.replace(/<ScrollView([^>]*?)>/g, (match, p1) => {
        if (!p1.includes('showsVerticalScrollIndicator')) {
            return `<ScrollView${p1} showsVerticalScrollIndicator={false} showsHorizontalScrollIndicator={false}>`;
        }
        return match;
    });

    newContent = newContent.replace(/<FlatList([^>]*?)(\/?)>/g, (match, p1, p2) => {
        if (!p1.includes('showsVerticalScrollIndicator')) {
            return `<FlatList${p1} showsVerticalScrollIndicator={false} showsHorizontalScrollIndicator={false}${p2}>`;
        }
        return match;
    });

    if (newContent !== content) {
        fs.writeFileSync(filepath, newContent, 'utf8');
        console.log(`Updated: ${filepath}`);
    }
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts')) {
            processFile(fullPath);
        }
    }
}

for (const d of directories) {
    if (fs.existsSync(d)) walkDir(d);
}

console.log("Done");

File: remove_scrollbars.py
import os
import re

directories = ['app', 'components']

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    new_content = content
    
    # Process ScrollView
    new_content = re.sub(
        r'<ScrollView(?![^>]*showsVerticalScrollIndicator)', 
        r'<ScrollView showsVerticalScrollIndicator={false} showsHorizontalScrollIndicator={false}', 
        new_content
    )
    
    # Process FlatList
    new_content = re.sub(
        r'<FlatList(?![^>]*showsVerticalScrollIndicator)', 
        r'<FlatList showsVerticalScrollIndicator={false} showsHorizontalScrollIndicator={false}', 
        new_content
    )
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated: {filepath}")

for d in directories:
    for root, _, files in os.walk(d):
        for file in files:
            if file.endswith('.tsx') or file.endswith('.ts'):
                process_file(os.path.join(root, file))

print("Done")
